src/cuda/api/memory.hpp

```diff
@@ -222,6 +222,12 @@ inline void copy(void *destination, const void *source, size_t num_bytes)
 	}
 }
 
+/**
+ * Synchronously copies data between memory spaces and CUDA arrays.
+ *
+ * @param destination A CUDA array @ref cuda::array::array_t
+ * @param source A pointer to a a memory region of size `destination.size() * sizeof(T)`
```

Comment:
Then I might ask if the comment here is complete:

https://github.com/eyalroz/cuda-api-wrappers/blob/development/src/cuda/api/memory.hpp#L207

because the CUDA memory copy works with managed memory, pinned host memory, pageable host memory, and device memory

Reply:
My comment is not complete, and should be reworded.

Both of these functions can take pointers to plain vanilla global device memory. But I think they can also take pointers to constant memory (correct me if I'm wrong).

Reply:
Also... I know I made a lot of comments, and some of them are not entirely trivial to address, but - I'd rather you replied to them at a rate of more than one-at-atime or two-at-a-time...

Reply:
Well, never mind this for now. I've added a separate issue, #130, about reviewing these comments.